### PR TITLE
feat: add redo to streaming payment

### PR DESCRIPTION
**Pull request by CzarekDryl**
## Description

- Changed kebab menu options for streaming payment

## Testing

*  Install streaming payment extension
* Create streaming payment action
* In completed action screen open kebab menu
* Check if redo action works fine

## TODO

- Add redo action to activity table

Resolves https://github.com/JoinColony/colonyCDapp/issues/3141

### CODE REVIEW

File: src/components/v5/common/CompletedAction/partials/StreamingPayment/StreamingPayment.tsx [hunk 1]
```diff
@@ -127,19 +168,73 @@ const StreamingPayment: FC<StreamingPaymentProps> = ({
...
+  const isCustomInterval =
+    Number(interval) !== ONE_HOUR_IN_SECONDS &&
+    Number(interval) !== ONE_DAY_IN_SECONDS &&
+    Number(interval) !== ONE_DAY_IN_SECONDS * 7;
+  const startTimeDate = new Date(Number(startTime) * 1000);
+
+  const isStartImmediately =
+    Math.abs(startTimeDate.getTime() - new Date(createdAt).getTime()) < 60000;
+
+  // console.log({isStartImmediately, startTimeDate, createdAt});
```
Comment (iamsamgibbs) [inline src/components/v5/common/CompletedAction/partials/StreamingPayment/StreamingPayment.tsx:180]: Left over console log

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
Nice work on this! I tested every scenario with different start / end conditions I could think of and it is handling them all fine.

Screenshot 2024-11-13 at 14 40 47

Screenshot 2024-11-13 at 14 40 54

The only reason I'm rejecting, is because of this recent PR #3587 which re-enables the redo option from the activity table. Currently here it isn't possible to redo via the activity table. It would be great if the `feat/streaming-payments-ui` feature branch could be rebased onto master and make sure the redo via activity table functionality is working here. Especially as there is some custom logic in getting the correct values to show.

**Review by iamsamgibbs — APPROVED** [review]
Switching this to an approval and we can handle the redo via activity table as a separate issue after the branches have been rebased.

File: src/components/v5/common/ActionSidebar/hooks/useActionFormBaseHook.ts [hunk 2]
```diff
@@ -42,6 +42,7 @@ const useActionFormBaseHook: UseActionFormBaseHook = ({
     validationSchema,
     actionType,
     id,
+    transform,
```
Comment (bassgeta) [inline src/components/v5/common/ActionSidebar/hooks/useActionFormBaseHook.ts:45]: nice catch!

**Review by bassgeta — APPROVED** [review]
Works nice and the solution is very simple :+1: 
Tried out a few cases, got 2 on video too :eyes: 

https://github.com/user-attachments/assets/da782647-db5f-442b-aed6-29489192270d

**Review by rdig — CHANGES_REQUESTED** [review]
I've tested this but can only redo the streaming payment action from the sidebar menu, not from activity feed menu.

I believe redo-ing from the activity feed should also be included in this PR

From the sidebar:

Screenshot from 2024-11-22 18-24-51
Screenshot from 2024-11-22 18-25-53
image
Screenshot from 2024-11-22 18-26-39

From the activity feed:

Screenshot from 2024-11-22 18-27-00

**Comment by CzarekDryl** [comment]
@rdig Can we make it in other PR? `feat/streaming-payments` branch needs to be rebased to have changes from https://github.com/JoinColony/colonyCDapp/pull/3587

**Comment by rdig** [comment]
Sure, another PR is fine

**Review by rdig — APPROVED** [review]
All good here

**Pull request merged**